torchao/dtypes/nf4tensor.py

```diff
@@ -93,17 +319,36 @@ def mm_default(func, *args, **kwargs):
         aten.copy_.default,
     ]
 )
-def copy_(func, *args, **kwargs):
-    original: NF4Tensor = args[0][0]
-    copy_in: torch.Tensor = args[0][1]
-
-    # Base Case
+def nf4_copy_(aten_op, args, kwargs=None):
+    assert len(args) == 2 and (kwargs is None or len(kwargs) == 0), "only support aten.copy_.default with 2 args"
+    original: NF4Tensor = args[0]
+    copy_in: torch.Tensor = args[1]
 
     if same_metadata(original, copy_in):
-        original_tensors = original.__tensor_flatten__()[0]
-        for tensor_name in original_tensors:
-            getattr(original, tensor_name).copy_(getattr(copy_in, tensor_name))
-        return
+        quantized_scalers = aten_op(original.quantized_scalers, copy_in.quantized_scalers, **kwargs)
+        quantization_factor = aten_op(original.quantization_factor, copy_in.quantization_factor, **kwargs)
+        quantized_data = aten_op(original.quantized_data, copy_in.quantized_data, **kwargs)
+        scaler_mean = aten_op(original.scaler_mean, copy_in.scaler_mean, **kwargs)
+        nf4 = aten_op(original.nf4, copy_in.nf4, **kwargs)
+        tensor_meta = SubclassTensorArgs(
```

Comment:
This is also kinda confused me, like isn't this returning a new out of place tensors?

Reply:
you're right. I should just return `args[0] / self` instead